nautobot_ssot/jobs/base.py

```diff
@@ -47,17 +48,56 @@ class DataSyncBaseJob(BaseJob):
 
     dry_run = BooleanVar()
 
+    def load_source_adapter(self):
+        """TODO:."""
+        pass
+
+    def load_target_adapter(self):
+        """TODO:."""
+        pass
+
+    def calculate_diff(self):
+        """TODO:."""
+        pass
+
+    def execute_sync(self):
+        """TODO:."""
+        pass
+
     def sync_data(self):
         """Method to be implemented by data sync concrete Job implementations.
 
+        Note: You could completely overwrite it if desired.
+
         Available instance attributes include:
 
         - self.kwargs     (corresponds to the Job's `data` input, including 'dry_run' option)
         - self.commit     (should generally be True)
         - self.sync       (Sync instance tracking this job execution)
         - self.job_result (as per Job API)
         """
-        pass  # pylint: disable=unnecessary-pass
+        # TODO: most of the log messages from the methods could be standarised here
+        # example: self.log_info(message="Loading current data from source adapter...")
```

Comment:
I agree, these should be logged automatically in the default `sync_data` implementation.

Reply:
Could we store these datapoints in the Sync model ? it would be easier to query, analyze, compare and expose to other systems if needed

Reply:
good idea